plugins/workspace-backpack/src/backpackable.ts

```diff
@@ -0,0 +1,24 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly/core';
+
+/** Defines if an object can be added to the backpack. */
+export interface Backpackable {
+  /**
+   * Returns a representation of this object as a FlyoutItemInfo array, so that
+   * it can be displayed in the backpack.
+   *
+   * This method should remove any unique or useless state data (e.g. IDs or
+   * coordinates).
+   */
+  toFlyoutData(): Blockly.utils.toolbox.FlyoutItemInfo[];
+}
+
+/** Checks whether the given object conforms to the Backpackable interface. */
+export function isBackpackable(obj: any): obj is Backpackable {
```

Comment:
unknown might work here

Reply:
Unknown means you can't do anything with the value until you've asserted what type the value is. Importantly for this case, you can't dereference properties on them. So for type guards you have to use `any`.

https://devblogs.microsoft.com/typescript/announcing-typescript-3-0-rc-2/#the-unknown-type